File: src/tauri/commands/collect.ts
import { safeInvoke } from "./_base";

export interface CollectConfig {
  input: string;
  output: string;
  keyword: string;
  files: string;
  ext: string;
  limit_copy: number;
  skip_dir: string;
  no_default_skip: boolean;
  flat: boolean;
  group_by_parens: boolean;
  overwrite: boolean;
  create_history: boolean;
  delete_non_vn: boolean;
  report_dir: string;
  dry_run: boolean;
}

export interface CollectRunResult {
  ok: boolean;
  log: string[];
  summary: string;
}

export const DEFAULT_COLLECT_CONFIG: CollectConfig = {
  input: "",
  output: "",
  keyword: "",
  files: "",
  ext: "xlsx",
  limit_copy: 50,
  skip_dir: "",
  no_default_skip: false,
  flat: false,
  group_by_parens: false,
  overwrite: true,
  create_history: true,
  delete_non_vn: false,
  report_dir: "reports",
  dry_run: false,
};

export function collectLoadIni() {
  return safeInvoke<CollectConfig>("collect_load_ini");
}

export function collectRun(config: CollectConfig) {
  return safeInvoke<CollectRunResult>("collect_run", { config });
}

export function collectByFolders(config: CollectConfig) {
  return safeInvoke<CollectRunResult>("collect_by_folders", { config });
}

File: src-tauri/src/models/collect.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct CollectConfig {
    pub input: String,
    pub output: String,
    pub keyword: String,
    pub files: String,
    pub ext: String,
    pub limit_copy: i64,
    pub skip_dir: String,
    pub no_default_skip: bool,
    pub flat: bool,
    pub group_by_parens: bool,
    pub overwrite: bool,
    pub create_history: bool,
    pub delete_non_vn: bool,
    pub report_dir: String,
    pub dry_run: bool,
}

impl Default for CollectConfig {
    fn default() -> Self {
        CollectConfig {
            input: String::new(),
            output: String::new(),
            keyword: String::new(),
            files: String::new(),
            ext: "xlsx".into(),
            limit_copy: 50,
            skip_dir: String::new(),
            no_default_skip: false,
            flat: false,
            group_by_parens: false,
            overwrite: true,
            create_history: true,
            delete_non_vn: false,
            report_dir: "reports".into(),
            dry_run: false,
        }
    }
}

#[derive(Debug, Clone, Serialize)]
pub struct CollectRunResult {
    pub ok: bool,
    pub log: Vec<String>,
    pub summary: String,
}
